core/game_object.lua

```diff
@@ -1939,40 +1939,57 @@ Set `prefix_config.key = false` on your object instead.]]):format(obj.key), obj.
     ----- API CODE GameObject.PokerHand
     -------------------------------------------------------------------------------------------------
 
+    SMODS.PokerHandParts = {}
+    SMODS.PokerHandPart = SMODS.GameObject:extend {
+        obj_table = SMODS.PokerHandParts,
+        obj_buffer = {},
+        required_params = {
+            'key',
+            'func',
+        },
+        inject_class = function() end,
+    }
+    local handlist = G.handlist
+    G.handlist = {}
     SMODS.PokerHands = {}
     SMODS.PokerHand = SMODS.GameObject:extend {
         obj_table = SMODS.PokerHands,
         obj_buffer = {},
         required_params = {
             'key',
-            'above_hand',
             'mult',
             'chips',
             'l_mult',
             'l_chips',
             'example',
+            'evaluate'
         },
         order_lookup = {},
+        max_order = { value = 0 },
         visible = true,
         played = 0,
         played_this_round = 0,
         level = 1,
-        class_prefix = 'h',
         set = 'PokerHand',
         process_loc_text = function(self)
             SMODS.process_loc_text(G.localization.misc.poker_hands, self.key, self.loc_txt, 'name')
             SMODS.process_loc_text(G.localization.misc.poker_hand_descriptions, self.key, self.loc_txt, 'description')
         end,
         register = function(self)
             if self:check_dependencies() and not self.obj_table[self.key] then
-                local j
-                for i, v in ipairs(G.handlist) do
-                    if v == self.above_hand then j = i end
+                if self.above_hand and self.obj_table[self.above_hand] then
+                    local j
+                    for i, v in ipairs(G.handlist) do
+                        if v == self.above_hand then j = i end
+                    end
+                    table.insert(G.handlist, j, self.key)
+                    self.order_lookup[self.above_hand] = (self.order_lookup[self.above_hand] or 0) - 0.001
+                    self.order = self.obj_table[self.above_hand].order + self.order_lookup[self.above_hand]
+                else
+                    self.max_order.value = self.max_order.value + 1
+                    self.order = self.max_order.value
+                    G.handlist[#G.handlist+1] = self.key
```

Comment:
An idea I had for poker hands: auto-sort them based on the product of base mult/chips; above_hand (if set) should be considered first. If two hands are both set to be above the same hand, there could also be conflicts?

Reply:
I went ahead and added this, should also be more efficient than inserting each hand individually where it belongs. `order_offset` can be used to change ordering as needed. Multiple hands with the same above_hand are sorted the same way with lower priority